Update ReadME and Discord Help Message (#5)
SkyKings-Network/GuildBridgeBot: README.md

# GuildBridgeBot

> GuildBridgeBot is a powerful tool that bridges communication between Hypixel's Minecraft guilds and Discord, enabling seamless interaction between guild members and Discord users. This bot ensures real-time relay of all guild messages, notifications, and events, facilitating better communication and management.

Reviewer: ```suggestion
Guild Bridge Bot is a powerful tool that bridges communication between Hypixel's Minecraft guilds and Discord, 
enabling seamless interaction between guild members and Discord users. 
This bot ensures real-time relay of all guild messages, notifications, 
and events, facilitating better communication and management.
```

## Table of Contents
- [Features](#features)
- [Requirements](#requirements)
- [Installation and Usage](#installation-and-usage)
- [Commands](#commands)
- [Troubleshooting](#troubleshooting)
- [Contributing](#contributing)
- [License](#license)
- [Disclaimer](#disclaimer)
- [Support](#support)
- [To-Do List](#to-do-list)

## Features

- **Two-Way Communication**: Relay messages between Hypixel guild chat and a designated Discord channel.
- **Officer Chat Support**: Separate channel for officer communications with togglable modes.
- **Guild Event Notifications**: Automatically notify Discord about guild events such as:
  - Members joining or leaving
  - Promotions and demotions
  - Kicks
  - Mutes and unmutes
- **Command Handling**: Execute Hypixel guild commands from Discord.
- **Auto-Accept Invites**: Optionally auto-accept guild invites.
- **Robust Error Handling**: Automatic reconnection and error reporting.
- **Customizable Appearance**: Use Discord webhooks for more control over bot messages.

## Requirements

> - Python 3.10+
> - Discord.py library
> - Mineflayer (Node.js Minecraft client)
> - Redis (optional, for additional features)

Reviewer: ```suggestion
- Python 3.10+
    - discord.py
    - javascript
- node.js 
    - mineflayer
- Redis (optional, for additional features)
```

---

## Installation and Usage

### Initial Setup

> Follow these instructions to set up the bot on a Linux/Ubuntu machine:

Reviewer: ```suggestion
Follow these instructions to copy the repository to a new server.
```

| `!notifications` | Toggle join/leave notifications |
| `!toggleaccept` | Toggle auto-accepting guild invites |

### Miscellaneous

| Command | Description |
|---------|-------------|
| `!help` | Display all available commands and bot information |
| `!online` | Check online guild members |
| `!list` | Show a list of all guild members |
| `!override <command>` | Force the bot to use a given command |

Note: Some commands may require appropriate permissions in both Discord and the Hypixel guild.

---

## Troubleshooting

1. **Bot Not Starting:** 
   - Ensure the virtual environment is activated before running the bot. 
   - Verify that `pm2` is installed globally and the path to the Python interpreter is correct.

2. **Configuration Errors:** 
   - Double-check the `config.json` file for any misconfigurations or missing values.

3. **Slow Sync Issues:** 
   - For message sync delays, review the server and network speed.
   - Consider enabling Redis for better synchronization.

4. **Bot Disconnecting Frequently:**
   - Check the `config.json` file and ensure auto-reconnection settings are enabled.
   - Use Redis to prevent disconnections.

> If you need further assistance, please open an issue on the GitHub repository.

Reviewer: Can we possibly cut some or most of this out and link to the docs instead? this readme is getting long
